chia/plotting/manager.py

```diff
@@ -132,6 +182,19 @@ def _refresh_task(self):
                 self.log.debug(f"refresh_plots: Sleep {batch_sleep} milliseconds")
                 time.sleep(float(batch_sleep) / 1000.0)
 
+            # Cleanup unused cache
+            def plot_id_in_use(plot_id_in: bytes32):
+                for _, plot_info in self.plots.items():
```

Comment:
a linear search inside a linear search gives you `O(n^2)`. that seems risky for people with a lot of plots. I think you should put these IDs in a `set` up-front, and then look them up from there.

Reply:
Changed!